handler.py

```diff
@@ -0,0 +1,91 @@
+
+import os
+import json
+import urllib.parse
+
+from rdflib.plugins.sparql.parser import parseUpdate
+from rdflib.plugins.sparql.algebra import translateUpdate
+
+from sns_client import SnsClient
+from neptune_client import NeptuneClient
+
+
+def main(event, _):
+    rialto_sparql_endpoint = os.getenv('RIALTO_SPARQL_ENDPOINT', "http://localhost:8080/bigdata/namespace/kb/sparql")
+    rialto_sns_endpoint = os.getenv('RIALTO_SNS_ENDPOINT', "http://localhost:4575")
+    rialto_topic_arn = os.getenv('RIALTO_TOPIC_ARN', "rialto")
+    aws_region = os.getenv('AWS_REGION', "us-west-2")
+
+    sns_client = SnsClient(rialto_sns_endpoint, rialto_topic_arn, aws_region)
+    neptune_client = NeptuneClient(rialto_sparql_endpoint)
+
+    response, status_code = neptune_client.post(event['body'])
+
+    if status_code == 200:
+        if "update=" in event['body'] or event['Content-Type'] == "application/sparql-update":
+            entities = get_unique_subjects(
+                            get_entities(
+                                urllib.parse.unquote_plus(
+                                    event['body']).replace('update=', '')))
+            if entities:
```

Comment:
When is this if condition *not* triggered, and do we care about that condition?

Reply:
I doubt this will get triggered assuming any insert would have at least 1 subject, but I suspect it's a minor guard against their being issues sending an empty list to derivatives?

Reply:
OK, no biggie.